src/Elements/Geometry/Ray.cs

```diff
@@ -77,6 +78,102 @@ public bool Intersects(Triangle tri, out Vector3 result)
             }
         }
 
+        /// <summary>
+        /// Does this ray intersect with the provided SolidOperation?
+        /// </summary>
+        /// <param name="solidOp">The SolidOperation to intersect with.</param>
+        /// <param name="result">The intersection result.</param>
+        /// <returns>True if an intersection occurs, otherwise false. If true, check the intersection result for the location of the intersection.</returns>
+        public bool Intersects(SolidOperation solidOp, out Vector3 result)
+        {
+            var intersects = Intersects(solidOp.GetSolid(), out Vector3 tempResult);
+            result = tempResult;
+            return intersects;
+        }
+
+        /// <summary>
+        /// Does this ray intersect with the provided Solid? 
+        /// </summary>
+        /// <param name="solid">The Solid to intersect with.</param>
+        /// <param name="result">The intersection result.</param>
+        /// <returns>True if an intersection occurs, otherwise false. If true, check the intersection result for the location of the intersection.</returns>
+        internal bool Intersects(Solid solid, out Vector3 result)
+        {
+            var distanceToClosestResult = double.MaxValue;
+            var faces = solid.Faces;
+            var intersects = false;
+            var finalResult = default(Vector3);
+            foreach (var face in faces)
```

Comment:
As per the comment about, let's store all the intersections and return a `List<Vector3>` ordered by their proximity to the origin of the ray.

Reply:
Done.